File: PHP/Medium/HashTable/49__Group_Anagrams.php
<?php
//https://leetcode.com/problems/group-anagrams/description/

function groupAnagrams($strs)
{
    $anagrams = [];
    foreach ($strs as $str) {
        $key = str_split($str);
        sort($key);
        $key = implode("", $key);

        if (!isset($anagrams[$key])) {
            $anagrams[$key] = [];
        }
        $anagrams[$key][] = $str;
    }
    return array_values($anagrams);
}

$strs = ["eat", "tea", "tan", "ate", "nat", "bat"];
var_dump(groupAnagrams($strs));

File: Python/Medium/HashTable/49__Group_Anagrams.py
# https://leetcode.com/problems/group-anagrams/description/
from collections import defaultdict


def groupAnagrams(strs: list[str]) -> list[list[str]]:
    anagram = defaultdict(list)
    for str in strs:
        key = ''.join(sorted(str))
        anagram[key].append(str)
    return list(anagram.values())


strs = ["eat", "tea", "tan", "ate", "nat", "bat"]
print(groupAnagrams(strs))
